eng/Versions.props

```diff
@@ -291,24 +292,21 @@
       create a test insertion in Visual Studio to validate.
     -->
     <NewtonsoftJsonVersion>13.0.1</NewtonsoftJsonVersion>
-    <StreamJsonRpcVersion>2.14.20</StreamJsonRpcVersion>
+    <StreamJsonRpcVersion>2.15.4-alpha</StreamJsonRpcVersion>
     <!--
       When updating the S.C.I or S.R.M version please let the MSBuild team know in advance so they
       can update to the same version. Version changes require a VS test insertion for validation.
     -->
-    <SystemCollectionsImmutableVersion>6.0.0</SystemCollectionsImmutableVersion>
-    <SystemReflectionMetadataVersion>6.0.1</SystemReflectionMetadataVersion>
-    <MicrosoftBclAsyncInterfacesVersion>6.0.0</MicrosoftBclAsyncInterfacesVersion>
+    <SystemCollectionsImmutableVersion>7.0.0</SystemCollectionsImmutableVersion>
```

Comment:
The System.Collections.Immutable upgrade triggers this warning:

> /mnt/vss/_work/1/s/.packages/system.collections.immutable/7.0.0/buildTransitive/netcoreapp2.0/System.Collections.Immutable.targets(4,5): error : System.Collections.Immutable 7.0.0 doesn't support netcoreapp3.1 and has not been tested with it. Consider upgrading your TargetFramework to net6.0 or later. You may also set <SuppressTfmSupportBuildWarnings>true</SuppressTfmSupportBuildWarnings> in the project file to ignore this warning and attempt to run in this unsupported configuration at your own risk. [/mnt/vss/_work/1/s/src/Dependencies/Collections/Microsoft.CodeAnalysis.Collections.Package.csproj::TargetFramework=netcoreapp3.1]

I see only 3 project files that include netcoreapp3.1 as a target. Given this is no longer a supported runtime, shall I retarget these to net6.0 to resolve the warning?

![image](https://user-images.githubusercontent.com/3548/221636230-0d8e624f-8ebd-4f94-8fcc-daa1293cdea7.png)

Reply:
There really isn't much choiyce for us here, need to move forward here.

Reply:
As mentioned in https://github.com/dotnet/roslyn/pull/66561, the warning in MS.CA.Collections can be suppressed since it's a source package

Reply:
Wouldn't it be better to drop the declaration of support for a runtime version that Microsoft no longer supports, than to just suppress the warning that we're supporting something we shouldn't be?

Reply:
I'm fine wither way, give this is a source package and not published publicly afaik. FYI @sharwell @tmat is you have any good reason to stick with 3.1

Reply:
Since it's just a source package, I think we just need to keep netstandard2.0.

Reply:
However, this might be used for verifying we can successfully build against netcoreapp3.1. In that case we should keep the target and suppress the warning. I prefer not to drop support for something that works just because.

Reply:
I'll leave it to your team to decide.